METALIUM_GUIDE.md

```diff
@@ -114,6 +114,152 @@ TODO: Describe that TT wins at scale-out, best compute density at the server an
   <img width="1176" alt="image" src="https://github.com/tenstorrent/tt-metal/assets/3885633/d3c89155-6e4d-49cb-a95c-85654ac29e7d">
 <img width="1171" alt="image" src="https://github.com/tenstorrent/tt-metal/assets/3885633/73039d17-3bce-4ff5-b797-da1aa9b147c4">
 
+#### Behind the scenes, a Compute Kernel becomes Unpack, Math, Pack Kernels
+
+As the above single Tensix core architecture shows, the Compute Kernel uses
+three Baby RISCVs. tt-metal uses them to conduct unpack, math, and pack. It
+actually generates Unpack, Math, Pack kernels from a single Compute kernel.
+
+For details, let's check what tt-metal generates for the following compute
+kernel:
+```
+namespace NAMESPACE {
+void MAIN {
+  mm_init();
+  acquire_dst(tt::DstMode::Tile);
+
+  cb_wait_front(tt::CB::c_in0, /* number of tiles */ 1);
+  cb_wait_front(tt::CB::c_in1, /* number of tiles */ 1);
+
+  matmul_tiles(tt::CB::c_in0, tt::CB::c_in1, 0, 0, 0, false);
+
+  cb_pop_front(tt::CB::c_in1, /* number of tiles */ 1);
+  cb_pop_front(tt::CB::c_in0, /* number of tiles */ 1);
+
+  cb_reserve_back(tt::CB::c_out0, /* number of tiles */ 1);
+  pack_tile(0, tt::CB::c_out0);
+  cb_push_back(tt::CB::c_out0, /* number of tiles */ 1);
+
+  release_dst(tt::DstMode::Tile);
+}
+}  // namespace NAMESPACE
+```
+
+It takes two matrix tiles from `tt::CB::c_in0` and `tt::CB::c_in0` L1 and
```

Comment:
please also add that MATH and pACK synchronize via  tile registers (aka destination registers), and that `tile_regs_` APIs are used to acquire/commit on the math side, and wait/release on the pack side.